packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -495,6 +510,7 @@ export default class AwsS3Multipart extends BasePlugin {
         resumableUploads: true,
       },
     })
+    this.uppy.addPreProcessor(this.#setCompanionHeaders)
```

Comment:
I wonder why `setOptions` isn't enough and we need to add a pre-processor? I would think `setOptions` already updates the plugins state and uses it before `upload`?